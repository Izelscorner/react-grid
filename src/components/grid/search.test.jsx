import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Search from './search';

describe('<Search />', () => {
    configure({adapter: new Adapter()});
    let searchHandler;
    let wrapper;
  
    beforeAll(() => {
      searchHandler = jest.fn();
      wrapper = shallow(<Search searchHandler={searchHandler}  />);
    });
  
    it('matches snapshot', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  
    it('renders the input', () => {
      const input = wrapper.find('input');
      expect(input).toHaveLength(1);
    });
  
    it('calls handleSearch on type', function () {
      const search = wrapper.find('input');
      expect(search).toHaveLength(1);
      search.simulate('change',{target: {value: 'luke'}});
      expect(searchHandler).toHaveBeenCalledWith({target: {value: "luke"}});
    });
});
