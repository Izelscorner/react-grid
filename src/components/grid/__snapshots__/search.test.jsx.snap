// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> matches snapshot 1`] = `
<input
  className="form-control"
  id="search"
  name="search"
  onChange={[MockFunction]}
  placeholder="Search by name"
  type="text"
/>
`;
